src/test/java/TaskToEventTest.java

```diff
@@ -0,0 +1,157 @@
+
+public class TaskToEventTest {
+    TaskToEvent taskToEvent;
+    TaskInfo taskInfo;
+    EventScheduler scheduler;
+    CalendarManager manager;
+
+    @BeforeEach
+    void setup() {
+        taskToEvent = new TaskToEvent();
+        taskInfo = new MockTaskInfo();
+        manager = new MockCalendarManager();
+        scheduler = new EventScheduler(manager);
+    }
+
+    @Test
+    void getAvailableTime() {
+        LocalDateTime expected = scheduler.getAvailableTime(
+                new ArrayList<>() , taskInfo.getDuration()).truncatedTo(ChronoUnit.MINUTES);
+        LocalDateTime actual = taskToEvent.getAvailableTime(taskInfo, scheduler, new ArrayList<>());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void checkTimeAvailability() {
+        LocalDateTime userSuggestedTime = LocalDateTime.of(2021, 11, 26, 10, 0);
+        boolean expected = scheduler.isAvailable(userSuggestedTime.toLocalTime(), taskInfo.getDuration(),
+                userSuggestedTime.toLocalDate());
+        boolean actual = taskToEvent.checkTimeAvailability(taskInfo, scheduler, userSuggestedTime);
+        assertEquals(expected, actual);
+    }
+
+    private static class MockTaskInfo implements TaskInfo {
+
+        @Override
+        public long getId() {
+            return 207L;
+        }
+
+        @Override
+        public String getName() {
+            return "CSC207";
+        }
+
+        @Override
+        public Duration getDuration() {
+            return Duration.ofHours(2);
+        }
+
+        @Override
+        public LocalDateTime getDeadline() {
+            return LocalDateTime.of(2021, 11, 25, 12, 0);
+        }
+
+        @Override
+        public List<String> getSubtasks() {
+            List<String> subtasks = new ArrayList<>();
+            subtasks.add("Project");
+            subtasks.add("Quiz");
+            return subtasks;
+        }
+
+        @Override
+        public boolean getCompleted() {
+            return false;
+        }
+    }
+
+    private class MockCalendarManager implements CalendarManager {
+
+        @Override
+        public void addEvent(CalendarEventModel eventData) {
+
+        }
+
+        @Override
+        public void markEventAsCompleted(long eventId) {
+
+        }
+
+        @Override
+        public List<EventReader> getAllEvents() {
+            List<EventReader> events = new ArrayList<>();
+            events.add(new MockEventReader());
+            return events;
+        }
+
+        @Override
+        public void loadEvents(String filePath) throws IOException {
+
+        }
+
+        @Override
+        public void saveEvents(String savePath) throws IOException {
+
+        }
+    }
+
+    private class MockEventReader implements EventReader {
+
+        @Override
+        public long getId() {
+            return 0;
+        }
+
+        @Override
+        public String getName() {
+            return null;
+        }
+
+        @Override
+        public LocalTime getStartTime() {
```

Comment:
To test conflict, we should set some start and end time

Reply:
Conflict checking is EventScheduler's job, and the conflict testings are in EventSchedulerTest (since the purposes of TaskToEvent involves calling EventScheduler)

Reply:
Oh I see. You are correct.